Sources/IO/Geometry/GLTFImporter/index.js

```diff
@@ -1,26 +1,623 @@
+        const byteOffset = accessor.byteOffset + bufferView.byteOffset;
+        const numberOfComponents = getNumberOfComponentsForType(accessor.type);
+
+        primitive.attributeValues.position = extractDataArray(
+          buffer,
+          byteOffset,
+          bufferView,
+          accessor,
+          numberOfComponents
+        );
+      }
+    });
+  }
+
+  function extractPrimitiveAccessorData(primitive) {
+    if (primitive.indices >= 0) {
+      const accessor = model.GLTFData.accessors[primitive.indices];
+      const bufferView = model.GLTFData.bufferViews[accessor.bufferView];
+      if (accessor.type !== AccessorTypes.SCALAR) {
+        throw new Error(
+          "Invalid accessor.type value for primitive connectivity loading. Expected 'SCALAR'"
+        );
+      }
+      const buffer = model.GLTFData.buffers[bufferView.buffer].dataView;
+      const byteOffset = accessor.byteOffset + bufferView.byteOffset;
+
+      let byteStride = 0;
+      switch (accessor.componentType) {
+        case AccessorComponentTypes.UNSIGNED_BYTE:
+          byteStride = 1;
+          primitive.indicesArray = extractCellBufferDataUint8(
+            buffer,
+            byteOffset,
+            bufferView.byteStride ?? byteStride,
+            accessor.count,
+            primitive.cellSize,
+            primitive.mode
+          );
+          break;
+        case AccessorComponentTypes.UNSIGNED_SHORT:
+          byteStride = 2;
+          primitive.indicesArray = extractCellBufferDataUint16(
+            buffer,
+            byteOffset,
+            bufferView.byteStride ?? byteStride,
+            accessor.count,
+            primitive.cellSize,
+            primitive.mode
+          );
+          break;
+        case AccessorComponentTypes.UNSIGNED_INT:
+          byteStride = 4;
+          primitive.indicesArray = extractCellBufferDataUint32(
+            buffer,
+            byteOffset,
+            bufferView.byteStride ?? byteStride,
+            accessor.count,
+            primitive.cellSize,
+            primitive.mode
+          );
+          break;
+        default:
+          throw new Error(
+            'Invalid accessor.componentType for primitive connectivity. Expected either GL_UNSIGNED_BYTE, GL_UNSIGNED_SHORT or GL_UNSIGNED_INT.'
+          );
+      }
+    }
 
-  publicAPI.parse = (content) => {};
+    extractPrimitiveAttributes(primitive);
+  }
+
+  async function loadData() {
+    const promises = model.GLTFData.buffers.map((buffer, index) => {
+      let uri = buffer.uri;
+      // const length = buffer.byteLength;
+
+      if (!/^data:/.test(uri) && uri[0] !== '/') {
+        // relative path
+        uri = model.baseURL.concat('/', uri);
+      }
+
+      return fetchData(uri, { binary: true }).then((bufferContent) => {
+        const dataView = new DataView(bufferContent);
+        model.GLTFData.buffers[index].dataView = dataView;
+        return dataView;
+      });
+    });
+    await Promise.all(promises);
```

Comment:
add a comment to explain why you need to wait for all the fetches to be done before going to the next step